concepts/key-concepts.md

```diff
@@ -62,74 +66,79 @@ streams.
+The [Forward](../pipeline/inputs/forward.md) input plugin doesn't assign tags. This
+plugin speaks the Fluentd wire protocol called Forward where every Event already
+comes with a Tag associated. Fluent Bit will always use the incoming Tag set by the
+client.
 {% endhint %}
 
-A Tagged record must always have a Matching rule. To learn more about Tags and Matches check the [Routing](data-pipeline/router.md) section.
+A tagged record must always have a Matching rule. To learn more about Tags and
+Matches, see [Routing](data-pipeline/router.md).
 
 ## Timestamp
 
-The Timestamp represents the _time_ when an Event was created. Every Event contains a Timestamp associated. The Timestamp is a numeric fractional integer in the format:
+The timestamp represents the time an Event was created. Every Event contains a
+timestamp associated. A timestamp always exists, and is set by the Input plugin or
+discovered through a data parsing process.
+
+The timestamp is a numeric fractional integer in the format:
 
 ```javascript
 SECONDS.NANOSECONDS
 ```
 
-### Seconds
-
-It is the number of seconds that have elapsed since the _Unix epoch._
-
-### Nanoseconds
+where:
 
-Fractional second or one thousand-millionth of a second.
-
-{% hint style="info" %}
-A timestamp always exists, either set by the Input plugin or discovered through a data parsing process.
-{% endhint %}
+- `_Seconds_` is the number of seconds that have elapsed since the Unix epoch.
+- `_Nanoseconds_` is a fractional second or one thousand-millionth of a second.
 
 ## Match
 
-Fluent Bit allows to deliver your collected and processed Events to one or multiple destinations, this is done through a routing phase. A Match represent a simple rule to select Events where it Tags matches a defined rule.
+Fluent Bit lets you deliver your collected and processed Events to one or multiple
+destinations through a routing phase. A _Match_ represents a rule to select Events
+where a Tag matches a defined rule.
 
-To learn more about Tags and Matches check the [Routing](data-pipeline/router.md) section.
+To learn more about Tags and Matches, see [Routing](data-pipeline/router.md).
 
-## Structured Messages
+## Structured messages
 
-Source events can have or not have a structure. A structure defines a set of _keys_ and _values_ inside the Event message. As an example consider the following two messages:
+Source events can have a structure. A structure defines a set of `keys` and `values`
+inside the Event message to implement faster operations on data modifications.
+Fluent Bit handles every Event message as a structured message.
 
-### No structured message
+Consider the following two messages:
 
-```javascript
-"Project Fluent Bit created on 1398289291"
-```
-
-### Structured Message
+- No structured message
```

Comment:
```suggestion
- Unstructured message
```

Reply:
technically accurate, but I don't know if it's semantically accurate. It -has- a structure, it just doesn't have an additional imposed structure